types/src/storage.ts

```diff
@@ -0,0 +1,262 @@
+import {
+  Null,
+  u128,
+  u64,
+  Text,
+  Vec,
+  bool,
+  GenericAccountId as AccountId,
+  BTreeSet,
+  BTreeMap,
+  Option,
+} from '@polkadot/types'
+import { RegistryTypes } from '@polkadot/types/types'
+import { JoyBTreeSet, JoyEnum, JoyStructDecorated, WorkingGroup } from './common'
+import { MemberId } from './members'
+import { WorkerId } from './working-group'
+
+export class BalanceOf extends u128 {}
+export class DataObjectId extends u64 {}
+export class StorageBucketId extends u64 {}
+
+export type StorageBucketsPerBagValueConstraintType = {
+  min: u64
+  max_min_diff: u64
+}
+
+export class StorageBucketsPerBagValueConstraint
+  extends JoyStructDecorated({
+    min: u64,
+    max_min_diff: u64,
+  })
+  implements StorageBucketsPerBagValueConstraintType {}
+
+export type StorageDataObjectType = {
+  accepted: bool
+  deletion_prize: BalanceOf
+  size: u64
+}
+
+// DataObject from the storage pallet. Changed due to the types conflict resolution.
+export class StorageDataObject
+  extends JoyStructDecorated({
+    accepted: bool,
+    deletion_prize: BalanceOf,
+    size: u64,
+  })
+  implements StorageDataObjectType {}
+
+export class DataObjectIdSet extends JoyBTreeSet(DataObjectId) {}
+export class DistributionBucketId extends u64 {}
+export class DistributionBucketFamilyId extends u64 {}
+export class StorageBucketIdSet extends JoyBTreeSet(StorageBucketId) {}
+export class DistributionBucketSet extends JoyBTreeSet(DistributionBucketId) {}
+
+export type DynamicBagDeletionPrizeTypeDef = {
+  account_id: AccountId
+  prize: BalanceOf
+}
```

Comment:
Types like this are usually interfaces names `ITypeName`, so I think we should stick to this pattern:
```suggestion
interface IDynamicBagDeletionPrize {
  account_id: AccountId
  prize: BalanceOf
}
```

`TypeNameDef` is usually used for enum definition consts

Reply:
Fixed for all entries.